### PR TITLE
#4544 - Fix inadvertent whitespace in v1beta3 enablement

### PR DESCRIPTION
**Pull request by screeley44**
This was an attempt to fix an issue that I discovered when enabling v1beta3 in the apiserver config.  I accidentally had an extra trailing whitespace in the KUBE_API_ARGS and v1beta3 was never recognized. I tried to find a way to trim this value when it's loaded into the api server but was not successful.  So the proposed solution will avoid inadvertent white spaces in the --runtime_config= string.

### CODE REVIEW

File: pkg/master/server/server.go [hunk 1]
```diff
@@ -205,7 +205,18 @@ func (s *APIServer) Run(_ []string) error {
 		glog.Fatalf("Failure to start kubelet client: %v", err)
 	}
 
-	_, v1beta3 := s.RuntimeConfig["api/v1beta3"]
+        //#4544 - screeley44 - test
+        //avoid inadvertent white spaces in the apiserver config file
+        //i.e. KUBE_API_ARGS="--runtime_config=api/v1beta3  "
+        v1beta3 := false
+        for k, _ := range s.RuntimeConfig {
```
Comment (bgrant0607) [inline pkg/master/server/server.go]: Thanks, but this isn't the right "fix" for this problem. It needs to be addressed in pkg/util/configuration_map.go, by banning whitespace in key names.

**Comment by screeley44** [comment]
@bgrant0607 - yeah was looking at that originally and for some reason I could not strip any of the white space in the key names, let me try again and see if I am more successful, maybe I was stripping the value and not the key

**Comment by screeley44** [comment]
@bgrant0607 - so when I originally was messing with this on Friday, I wasn't trimming the else, I think this will solve the problem for efficiently.  Not sure how to update a Pull Request, is that even possible or do I need to create a new pull request?

func (m *ConfigurationMap) Set(value string) error {
        for _, s := range strings.Split(value, ",") {
                if len(s) == 0 {
                        continue
                }
                arr := strings.SplitN(s, "=", 2)
                if len(arr) == 2 {
                        (*m)[strings.TrimSpace(arr[0])] = strings.TrimSpace(arr[1])
                } else {
                        (*m)[strings.TrimSpace(arr[0])] = ""
                }
        }
        return nil
}

**Comment by bgrant0607** [comment]
Pushing to the same branch updates the PR automatically. Will take a look, thanks.

File: pkg/master/server/server.go [hunk 2]
```diff
@@ -205,7 +205,8 @@ func (s *APIServer) Run(_ []string) error {
 		glog.Fatalf("Failure to start kubelet client: %v", err)
 	}
 
-	_, v1beta3 := s.RuntimeConfig["api/v1beta3"]
+        _, v1beta3 := s.RuntimeConfig["api/v1beta3"]
```
Comment (bgrant0607) [inline pkg/master/server/server.go]: You introduced a space here by accident.

**Comment by bgrant0607** [comment]
LGTM.

**Comment by bgrant0607** [comment]
Please fix the space and squash your commits.

**Comment by bgrant0607** [comment]
Thanks. Have you completed the CLA, or do you have a corporate CLA? If the latter, what company are you with?

**Comment by screeley44** [comment]
@bgrant0607 - I signed the individual CLA, not sure about the corporate CLA, but I'm with Redhat.
